api/api.go

```diff
@@ -458,9 +460,104 @@ func setExtensionUrls(original core.Extension, rootUrl string) core.Extension {
 		}
 	}
 
+	localization, err := GetLocalization(&extension)
+	if err != nil {
+		log.Printf("failed to retrieve locales: %v", err)
+	}
+
+	extension.Localization = localization
+
 	return extension
 }
 
+func GetLocalization(extension *core.Extension) (*core.Localization, error) {
+	path := filepath.Join(".", extension.Development.RootDir, "locales")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// The extension does not have a locales directory.
+		return nil, nil
+	}
+
+	fileNames, err := GetFileNames(path)
+	if err != nil {
+		return nil, err
+	}
+	translations := make(map[string]interface{})
+	defaultLocale := ""
+
+	for _, fileName := range fileNames {
+		data, err := GetMapFromFile(filepath.Join(path, fileName))
+		if err != nil {
+			return nil, err
+		}
+
+		locale := strings.Split(fileName, ".")[0]
+
+		if IsDefaultLocale(fileName) {
+			defaultLocale = locale
+		}
+
+		translations[locale] = data
+	}
+
+	if len(translations) == 0 {
+		return nil, nil
+	} else {
+		if defaultLocale == "" {
+			log.Println("could not determine a default locale, please ensure you have a {locale}.default.json file")
+		}
+
+		return &core.Localization{
+			DefaultLocale: defaultLocale,
+			Translations:  translations,
+		}, nil
+	}
+}
+
+func GetFileNames(folderPath string) ([]string, error) {
+	files := []string{}
+	items, err := ioutil.ReadDir(folderPath)
```

Comment:
Is there a reason to use this instead of `os.ReadDir`. From the [docs](https://pkg.go.dev/io/ioutil#pkg-overview) it looks like `os` is the preferred package

> Package ioutil implements some I/O utility functions.
> 
> As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details.

Reply:
Looks like you are correct, we should be using `os` then. Will update that 👍